### PR TITLE
Don't insert * after new line in multi-line comment if previous line …

### LINKED ISSUE #3386: Don't insert * on comment continuation if previous line doesn't start with *
Suppose I need to add text to a comment written in the following style:
```
/*
First line
Second line
Third line
*/
```
When I add a new line, it should be clear that I don't want automatic `* ` since the previous line doesn't have it, yet Geany still inserts it.

### PR DESCRIPTION
**Pull request by ntrel**
…doesn't start with *

Fixes #3386.

Note: This was already the case when the previous line is blank, but a * was inserted when it is not blank.

### CODE REVIEW

File: src/editor.c [hunk 1]
```diff
@@ -3526,10 +3526,13 @@ static void auto_multiline(GeanyEditor *editor, gint cur_line)
 		{ /* we are on the second line of a multi line comment, so we have to insert white space */
 			whitespace = " ";
 		}
-
-		if (style == SCE_D_COMMENTNESTED)
-			continuation = "+"; /* for nested comments in D */
-
+		else if (!(g_str_has_prefix(previous_line + i, continuation) &&
```
Comment (techee) [inline src/editor.c:3529]: I'd just suggest inverting the condition (i.e. removing the initial `!`) and adding the 3 lines after this block inside the block. This way you avoid the duplicated `g_free(previous_line);` and the extra `return`.

Comment (techee) [reply src/editor.c:3529]: Apart from that the PR looks fine to me and is useful, I also did run into this issue in the past.

Comment (ntrel) [reply src/editor.c:3529]: That would break the first if clause before the else if on this line.

Comment (techee) [reply src/editor.c:3529]: Ah, right, sorry, I didn't think about that.

Comment (ntrel) [reply src/editor.c:3529]: No problem, thanks for reviewing.

File: doc/geany.txt [hunk 2]
```diff
@@ -2208,7 +2208,8 @@ Automatic continuation multi-line comments
       *
 
     on the next line with the correct indentation based on the previous line,
-    as long as the multi-line is not closed by ``*/``.
+    as long as the multi-line is not closed by ``*/``. If the previous line
+    has no ``*`` prefix, the new line will not be changed.
```
Comment (elextr) [inline doc/geany.txt:2212]: Suggest "the new line will not be changed" s/b "no ``*`` will be added to the new line" just to be explicit

Comment (ntrel) [reply doc/geany.txt:2212]:
```suggestion
    has no ``*`` prefix, no ``*`` will be added to the new line.
```

**Review by b4n — APPROVED** [review]
Looks good and works well :+1:

**Pull request merged**